xdsl/dialects/riscv.py

```diff
@@ -451,7 +447,9 @@ def __init__(
         comment: str | StringAttr | None = None,
     ):
         if isinstance(immediate, int):
-            immediate = SImm12Attr(immediate)
+            immediate = SImm12Attr(
+                immediate
+            )  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
@adutilleul can you please copy/paste the error you get when you remove the pyright ignore?
@math-fehr do you know why we might need a type ignore here?

Reply:
`No overloads for "__new__" match the provided arguments
  Argument types: (int)Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)`
I tried to replicate the sort of "hack" that was done on IntegerAttr to fix the pyright issue but not luck so far.

Reply:
@superlopuh, any idea how to resolve this?

Reply:
I added another overload to IntAttr to accept no args to __new__. Would be good to have a tick from @math-fehr  before merging